scripts/standby_node_health_check/standby_node_health_check.py

```diff
@@ -0,0 +1,530 @@
+            if cur_line_match:
+                mlx_port = cur_line_match.group(1)
+                ib_port = cur_line_match.group(2)
+                ib_to_mlx_map[ib_port] = mlx_port
+        return ib_to_mlx_map
+
+    @classmethod
+    def _check_ib_interface(cls, ib_interface: str, ib_interfaces_status: dict):
+        if ib_interface in ib_interfaces_status:
+            ib_state = ib_interfaces_status[ib_interface]["State"]
+            physical_state = ib_interfaces_status[ib_interface]["Physical_state"]
+            if ib_state == "active" and physical_state == "linkup":
+                return True
+        return False
+
+    @classmethod
+    def _run_and_parse_ibstat(cls):
+        retcode, stdout = cls._run_command(cls.IBSTAT_COMMAND)
+        if retcode != 0:
+            logger.error("Cannot run ibstat")
+            return {}
+        lines = stdout.splitlines()
+        ca_info = {}
+        current_ca = None
+
+        for line in lines:
+            line = line.strip()
+            if line.startswith("CA '"):
+                current_ca = line.split("'")[1]
+                ca_info[current_ca] = {}
+            elif current_ca and line.startswith("State:"):
+                ca_info[current_ca]["State"] = line.split(":")[1].strip().lower()
+            elif current_ca and line.startswith("Physical state:"):
+                ca_info[current_ca]["Physical_state"] = (
+                    line.split(":")[1].strip().lower()
+                )
+
+        return ca_info
+
+    @classmethod
+    def _run_and_parse_ip_link_show(cls):
+        ret_code, ip_link_show_output = cls._run_command(cls.IP_LINK_SHOW_COMMAND)
+        if ret_code != 0:
+            logger.error("Failed to run ip link show")
+            return {}
+        interfaces = cls._parse_ip_link_output(ip_link_show_output)
+        return interfaces
+
+    @classmethod
+    def _parse_ip_link_output(cls, ip_link_output: str):
+        interfaces = {}
+        try:
+            link_data = json.loads(ip_link_output)
+            for entry in link_data:
+                ifname = entry.get("ifname")
+                operstate = entry.get("operstate", "").lower()  # Get state in lowercase
+                link_type = entry.get("link_type", "").lower()
+
+                if ifname and link_type == "ether":
+                    interfaces[ifname] = operstate
+
+        except json.JSONDecodeError:
+            logger.error("Error while parssing ib link show command")
+        return interfaces
+
+    def _check_mgmt_interface(self):
+        interfaces_status = self._run_and_parse_ip_link_show()
+        result = True
+        if self._mgmt_interface not in interfaces_status:
+            logger.warning(
+                "%s is not in the list of management interfaces", self._mgmt_interface
+            )
+            result = False
+        elif interfaces_status[self._mgmt_interface] != "up":
+            logger.warning(
```

Comment:
what happen if all management interfaces are "down" - shouldn't we get error log message and exit?

Reply:
same answer as above - we decided to do all the tests no matter what. And for now, only give a result per each input interface